src/prog_algs/state_estimators/particle_filter.py

```diff
@@ -9,6 +9,8 @@
 from ..exceptions import ProgAlgTypeError
 from warnings import warn
 
+import matplotlib.pyplot as plt
+import numpy as np
```

Comment:
I don't think these are needed here for your change. Can you remove these from the PR?